WordPress/Classes/ViewRelated/Blog/SiteManagement/SiteTagsViewController.swift

```diff
@@ -0,0 +1,376 @@
+                                                            target: self,
+                                                            action: #selector(createTag))
+    }
+
+    private func setupSearchBar() {
+       tableView.tableHeaderView = searchController.searchBar
+    }
+
+    private func removeSearchBar() {
+        tableView.tableHeaderView = nil
+    }
+
+    @objc private func createTag() {
+        navigate(to: nil)
+    }
+
+    private func refreshNoResultsView() {
+        guard resultsController.fetchedObjects?.count == 0 else {
+            noResultsView.removeFromSuperview()
+            setupSearchBar()
+            tableView.reloadData()
+            return
+        }
+
+        if isPerformingInitialSync {
+            setupLoadingView()
+        } else {
+            setupEmptyResultsView()
+        }
+
+        if noResultsView.superview == nil {
+            tableView.addSubview(withFadeAnimation: noResultsView)
+        }
+
+        removeSearchBar()
+    }
+
+    private func setupLoadingView() {
+        noResultsView.accessoryView = nil
+        noResultsView.titleText = loadingMessage()
+        noResultsView.messageText = ""
+        noResultsView.buttonTitle = ""
+    }
+
+    private func setupEmptyResultsView() {
+        noResultsView.accessoryView = noResultsAccessoryView()
+        noResultsView.titleText = noResultsTitle()
+        noResultsView.messageText = noResultsMessage()
+        noResultsView.buttonTitle = noResultsButtonTitle()
+        noResultsView.button.addTarget(self, action: #selector(createTag), for: .touchUpInside)
+    }
+
+    func tagsFailedLoading(error: Error) {
+        DDLogError("Tag management. Error loading tags for \(String(describing: blog.url)): \(error)")
+    }
+}
+
+// MARK: - Table view datasource
+extension SiteTagsViewController {
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return resultsController.sections?.count ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return resultsController.sections?[section].numberOfObjects ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: TableConstants.cellIdentifier, for: indexPath) as? TitleBadgeDisclosureCell, let tag = tagAtIndexPath(indexPath) else {
+            return TitleBadgeDisclosureCell()
+        }
+
+        cell.name = tag.name
+
+        if let count = tag.postCount?.intValue, count > 0 {
+            cell.count = count
+        }
+
+        return cell
+    }
+
+    fileprivate func tagAtIndexPath(_ indexPath: IndexPath) -> PostTag? {
+        return resultsController.object(at: indexPath) as? PostTag
+    }
+
+    func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
```

Comment:
This is a method from `NSFetchedResultsControllerDelegate`, but it's in the section marked `Table view datasource`. Thoughts on reorganizing this a bit to clean up the layout of these methods?